openpype/hosts/nuke/plugins/publish/validate_rendered_frames.py

```diff
@@ -14,35 +14,44 @@ def get_instance(context, plugin):
         # Get the errored instances
         return get_errored_instances_from_context(context, plugin=plugin)
 
-    def repair_knob(self, instances, state):
+    def repair_knob(self, context, instances, state):
+        create_context = context.data["create_context"]
         for instance in instances:
-            node = instance.data["transientData"]["node"]
-            files_remove = [os.path.join(instance.data["outputDir"], f)
-                            for r in instance.data.get("representations", [])
-                            for f in r.get("files", [])
-                            ]
+            files_remove = [
+                os.path.join(instance.data["outputDir"], f_)
+                for r_ in instance.data.get("representations", [])
+                for f_ in r_.get("files", [])
+            ]
             self.log.info("Files to be removed: {}".format(files_remove))
-            for f in files_remove:
-                os.remove(f)
-                self.log.debug("removing file: {}".format(f))
-            node["render"].setValue(state)
+            for f_ in files_remove:
+                os.remove(f_)
+                self.log.debug("removing file: {}".format(f_))
```

Comment:
I wouldn't want it to delete any files on disk just when clicking 'repair' to set the render mode. Isn't Nuke capable of just overwriting existing files?  Or does publishing in Nuke work in such a way that it only ever publishes existing frames from disk?

Reply:
I agree. The leftover code seems unrelated to the validator's purpose. Thanks for the fresh perspective. I hadn't considered it before.